### PR TITLE
feat: config-provider support divider className and style properties

### PR DESCRIPTION
**Pull request by MuxinFeng**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/discussions/39862

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     `ConfigProvider` support `Divider` className and style properties      |
| 🇨🇳 Chinese |      `ConfigProvider` 支持 `Divider` 组件的 className 和 style 属性     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 0095c45</samp>

This pull request adds a new `divider` prop to the `ConfigProvider` component, which enables global customization of the `Divider` component's appearance and behavior. It also updates the API documentation, adds tests, and fixes some minor issues in the code.

### <samp>🤖 Generated by Copilot at 0095c45</samp>

*  Add the `divider` prop to the `ConfigProvider` component, which can set the common props of the `Divider` component ([link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR5), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR91-R94), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R10), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R140-R143), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R231), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R281), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R73), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R74))
*  Fix a typo in the API documentation of the `ConfigProvider` component, where the `button` prop was incorrectly described as setting the `Select` common props ([link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563L66-R66))
*  Break a long comment line into two lines in the `ConfigProvider` component for readability and code style ([link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56L41-R43))
*  Add the `divider` prop to the `Divider` component's context consumption, props destructuring, class name logic, and style logic, which allow it to use the custom props from the `ConfigProvider` component if provided ([link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5L22-R22), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5R34), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5L56-R57), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-47eacd5e3519051f10cac0341c498fc5de5a9a88faf3c9356aeed90a3ec66cf5L85-R86))
*  Import the `Divider` component and the `DividerProps` type from the `divider` module in the `ConfigProvider` component and its tests, which are needed for the new prop and the new test cases ([link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5R8), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR5), [link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R10))
*  Add two new test cases to check that the `ConfigProvider` component can pass the `className` and `style` props to the `Divider` component through the `divider` prop ([link](https://github.com/ant-design/ant-design/pull/43042/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5R202-R232))

### CODE REVIEW

File: components/config-provider/context.ts [hunk 1]
```diff
@@ -87,6 +88,10 @@ export interface ConfigConsumerProps {
     showSearch?: boolean;
   };
   button?: ButtonConfig;
+  divider?: {
+    className?: DividerProps['className'];
+    style?: DividerProps['style'];
```
Comment (MadCcc) [inline components/config-provider/context.ts:93]:
```suggestion
    className?: string;
    style?: React.CSSProperties;
```
这个一般不会有变化

File: components/divider/index.tsx [hunk 2]
```diff
@@ -53,7 +54,7 @@ const Divider: React.FC<DividerProps> = (props) => {
       [`${prefixCls}-no-default-orientation-margin-left`]: hasCustomMarginLeft,
       [`${prefixCls}-no-default-orientation-margin-right`]: hasCustomMarginRight,
     },
-    className,
+    className ?? divider?.className,
```
Comment (MadCcc) [inline components/divider/index.tsx:57]: 这里应该不是条件判断哦，应该都可以生效

File: components/divider/index.tsx [hunk 3]
```diff
@@ -82,7 +83,7 @@ const Divider: React.FC<DividerProps> = (props) => {
   }
 
   return wrapSSR(
-    <div className={classString} {...restProps} role="separator">
+    <div className={classString} style={style ?? divider?.style} {...restProps} role="separator">
```
Comment (MadCcc) [inline components/divider/index.tsx:86]: 同上，这里需要合并

File: components/divider/index.tsx (hunk 2 above)
Comment (MadCcc) [reply components/divider/index.tsx:57]: 改

File: components/config-provider/index.en-US.md [hunk 4]
```diff
@@ -63,13 +63,14 @@ Some components use dynamic style to support wave effect. You can config `csp` p
...
+| button | Set Button common props | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - | 5.6.0 |
 | locale | Language package setting, you can find the packages in [antd/locale](http://unpkg.com/antd/locale/) | object | - |  |
 | prefixCls | Set prefix className | string | `ant` |  |
 | renderEmpty | Set empty content of components. Ref [Empty](/components/empty/) | function(componentName: string): ReactNode | - |  |
 | space | Set Space common props, ref [Space](/components/space) | { size: `small` \| `middle` \| `large` \| `number`, className?: string, style?: React.CSSProperties, classNames?: { item: string }, styles?: { item: React.CSSProperties } } | - | 5.6.0 |
 | theme | Set theme, ref [Customize Theme](/docs/react/customize-theme) | - | - | 5.0.0 |
 | virtual | Disable virtual scroll when set to `false` | boolean | - | 4.3.0 |
+| divider | Set Divider common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
```
Comment (MadCcc) [inline components/config-provider/index.en-US.md:73]: 这个顺序排一下吧

File: components/config-provider/index.zh-CN.md [hunk 5]
```diff
@@ -50,24 +50,25 @@ export default Demo;
...
 | getTargetContainer | 配置 Affix、Anchor 滚动监听容器。 | () => HTMLElement | () => window | 4.2.0 |
 | iconPrefixCls | 设置图标统一样式前缀 | string | `anticon` | 4.11.0 |
 | input | 设置 Input 组件的通用属性 | { autoComplete?: string } | - | 4.2.0 |
-| select | 设置 Select 组件的通用属性 | { showSearch?: boolean } | - |  |
-| button | 设置 Button 组件的通用属性 | { className?: string, style?: React.CSSProperties, classNames?: { icon: string }, styles?: { icon: React.CSSProperties } } | - | 5.6.0 |
 | locale | 语言包配置，语言包可到 [antd/locale](http://unpkg.com/antd/locale/) 目录下寻找 | object | - |  |
-| prefixCls | 设置统一样式前缀 | string | `ant` |  |
+| popupMatchSelectWidth | 下拉菜单和选择器同宽。默认将设置 `min-width`，当值小于选择框宽度时会被忽略。`false` 时会关闭虚拟滚动 | boolean \| number | - | 5.5.0 |
+| popupOverflow | Select 类组件弹层展示逻辑，默认为可视区域滚动，可配置成滚动区域滚动 | 'viewport' \| 'scroll' <InlinePopover previewURL="https://user-images.githubusercontent.com/5378891/230344474-5b9f7e09-0a5d-49e8-bae8-7d2abed6c837.png"></InlinePopover> | 'viewport' | 5.5.0 |
+| prefixCls | 设置统一样式前缀 | string | `ant` |  |  | renderEmpty | 自定义组件空状态。参考 [空状态](/components/empty-cn) | function(componentName: string): ReactNode | - |  |
```
Comment (MadCcc) [inline components/config-provider/index.zh-CN.md:69]:
```suggestion
| prefixCls | 设置统一样式前缀 | string | `ant` |  |  
| renderEmpty | 自定义组件空状态。参考 [空状态](/components/empty-cn) | function(componentName: string): ReactNode | - |  |
```

**Review by MadCcc — APPROVED** [review]

File: components/config-provider/index.zh-CN.md [hunk 6]
```diff
@@ -50,28 +50,28 @@ export default Demo;
...
 | renderEmpty | 自定义组件空状态。参考 [空状态](/components/empty-cn) | function(componentName: string): ReactNode | - |  |
+| select | 设置 Select 组件的通用属性 | { showSearch?: boolean } | - |  |
 | space | 设置 Space 的通用属性，参考 [Space](/components/space-cn) | { size: `small` \| `middle` \| `large` \| `number`, className?: string, style?: React.CSSProperties, classNames?: { item: string }, styles?: { item: React.CSSProperties } } | - | 5.6.0 |
 | theme | 设置主题，参考 [定制主题](/docs/react/customize-theme-cn) | - | - | 5.0.0 |
 | virtual | 设置 `false` 时关闭虚拟滚动 | boolean | - | 4.3.0 |
-| divider | 设置 Divider 组件的通用属性 | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
```
Comment (afc163) [inline components/config-provider/index.zh-CN.md]: 组件属性是不是单开个 Table 来写？

File: components/config-provider/index.en-US.md [hunk 7]
```diff
@@ -102,6 +97,17 @@ const {
 | componentDisabled | antd component disabled state | boolean | - | 5.3.0 |
 | componentSize | antd component size state | `small` \| `middle` \| `large` | - | 5.3.0 |
 
+### ComponentConfig
```
Comment (MadCcc) [inline components/config-provider/index.en-US.md:100]:
```suggestion
### Component Config
```

**Pull request merged**